File: set_1/7.rb
#!/usr/bin/env ruby

require 'base64'
require 'openssl'

# Read ciphertext, convert from base64
ciphertext = File.open('7-ciphertext.txt', 'rb').read
cipherbytes = Base64.decode64(ciphertext)

# Decrypt
key = 'YELLOW SUBMARINE'

aes_cipher = OpenSSL::Cipher.new('aes-128-ecb')
aes_cipher.decrypt
aes_cipher.key = key

plaintext = aes_cipher.update(cipherbytes) + aes_cipher.final

puts plaintext

File: set_1/7.py
#!/usr/bin/env python3

import codecs

from Crypto.Cipher import AES

def main():
    cipherBytes = bytes()
    with open('7-ciphertext.txt', 'rb') as ciphertextFile:
        ciphertext = ciphertextFile.read()
    cipherBytes = base64ToBytes(ciphertext)

    key = 'YELLOW SUBMARINE'
    cipher = AES.new(key, AES.MODE_ECB)
    msg = cipher.decrypt(cipherBytes)
    print(msg)

def base64ToBytes(the_base64):
    return codecs.decode(the_base64, 'base64')

if __name__ == '__main__':
    main()

